src/cargo/sources/git/utils.rs

```diff
@@ -889,7 +889,7 @@ pub fn fetch(
                             refspecs.iter().map(|s| s.as_str()),
                             git::remote::Direction::Fetch,
                         )?
-                        .connect(git::remote::Direction::Fetch, progress.add_child("fetch"))?
+                        .connect(git::remote::Direction::Fetch, &mut progress)?
```

Comment:
Great catch!

Something I tend to forget is that the progress hierarchy is limited in depth. At the time, 4 levels could be used effectively. After that, it would just start to clobber other progress bars which led to me being unable to find it and progress would just be missing.

This was fixed by first increasing the the maximum usable depth to 6 levels, and since the added `"fetch"` child didn't serve any purpose, it was removed as well.

As a `prodash` user the one using the `Progress` trait will be tempted to 'design' progress information so that everything is represented neatly, modelling the actual structure of the computation. I think a good example for this is `gix clone <repo big enough to see progress>`.